src/adapter/Cargo.toml

```diff
@@ -15,6 +15,7 @@ arrow = { version = "51.0.0", default-features = false }
 async-trait = "0.1.68"
 bytes = "1.3.0"
 bytesize = "1.1.0"
+cidr = "0.2.3"
```

Comment:
Would prefer to use the [`ipnet`](https://docs.rs/ipnet/2.10.0/ipnet/) crate instead. It's the gold standard in the Rust ecosystem for CIDR types AFAIK, and is already in our tree in this repo and in the cloud repo. It should offer all the APIs you're using from the `cidr` crate, plus many more.